mmv1/products/certificatemanager/TrustConfig.yaml

```diff
@@ -0,0 +1,120 @@
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      A user-defined name of the trust config. Trust config names must be unique globally.
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    description: |
+      The trust config location.
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      The creation timestamp of a TrustConfig.
+
+      A timestamp in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to nine fractional digits.
+      Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      The last update timestamp of a TrustConfig.
+
+      A timestamp in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to nine fractional digits.
+      Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+    output: true
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: 'labels'
+    description: 'Set of label tags associated with the trust config.'
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      One or more paragraphs of text description of a trust config.
+  - !ruby/object:Api::Type::Array
+    name: trustStores
+    description: |
+      Set of trust stores to perform validation against.
+      This field is supported when TrustConfig is configured with Load Balancers, currently not supported for SPIFFE certificate validation.
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::Array
+          name: trustAnchors
+          description: |
+            List of Trust Anchors to be used while performing validation against a given TrustStore.
+          item_type: !ruby/object:Api::Type::NestedObject
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'pemCertificate'
+                description: |
+                  PEM root certificate of the PKI used for validation.
+                  Each certificate provided in PEM format may occupy up to 5kB.
+                sensitive: true
+        - !ruby/object:Api::Type::Array
+          name: intermediateCas
+          description: |
+            Set of intermediate CA certificates used for the path building phase of chain validation.
+            The field is currently not supported if trust config is used for the workload certificate feature.
+          item_type: !ruby/object:Api::Type::NestedObject
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'pemCertificate'
+                description: |
+                  PEM intermediate certificate used for building up paths for validation.
+                  Each certificate provided in PEM format may occupy up to 5kB.
+                sensitive: true
```

Comment:
It looks like these fields are updatable, would it be possible to add a handwritten update test?

Reply:
@ScottSuarez Absolutely, that said I am not entirely sure how to test the update of the certificates.
We only have one certificate within the test fixture folder, I see that for other resources that use certificates we also do not have an update case.

Is it possible to add an additional test certificate upstream so I can use those in my tests?

Reply:
Never mind, I thought these test fixtures didn't come from magic-modules but they do. I will add the additional tests shortly.

Reply:
@ScottSuarez I am happy that you asked to add an update test because there were some challenges.

I have an odd case here, it seems that the only updateMask parameter this patch call accepts is "*"
I created a `pre-update` and set this as an parameter but I am not sure if this was the right approach.

What do you think is the right approach in this case?

```---[ REQUEST ]---------------------------------------
PATCH /v1/projects/project-x/locations/us-central1/trustConfigs/tf-test-trust-configiiycziq1cd?alt=json&updateMask=description HTTP/1.1
Host: certificatemanager.googleapis.com
User-Agent: Terraform/1.5.0 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google/dev
Content-Length: 2714
Content-Type: application/json
Accept-Encoding: gzip

{
 "description": "sample description for the trust config 2",
 "labels": {
  "bar": "foo"
 },
 "trustStores": [
  {
   "intermediateCas": [
    {
     "pemCertificate": "-----BEGIN CERTIFICATE-----\nMIIDazCCAlOgAwIBAgIUfCJocHZRYbgzWUuaP6j9HM4Fp9YwDQYJKoZIhvcNAQEL\nBQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM\nGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMzAzMjExNjM2MDFaFw0yNTEy\nMTYxNjM2MDFaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw\nHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggEiMA0GCSqGSIb3DQEB\nAQUAA4IBDwAwggEKAoIBAQC/XVbAB3l6cnizpofzSJ5o4Iko3WgNV1PNeTe/JvZP\nzSoJC22TY3foqrtM0YBdNWfBEKyT5bTGYLJ67P9v2CX4xfUtQHiuWkF1OaA8/umb\nvzaQt2V1XsPC4nlnByeL8nT9wtLEwZ/8qeYJ0c0PS+1zJ5kEVRCvTg3ay6o7VZv5\nt3TvaQBjNwAhaUlFw9R6tqCG7Ztt+4d8dxCV4LjC6ygHtzjfmJQJBgMJcCB3F0ch\nFMGFz9I4OGXrqukzuSJLMg/Nok3EkOV7b4vptdICfzJJTDfvyQ9radzC+O7mtnRp\nuYGdTFKbDOglwxliyJJuQ+GAWjaW2sM51EM6AIk88bylAgMBAAGjUzBRMB0GA1Ud\nDgQWBBT9f1xhI06cdr9fhizBFmmQlQQN0DAfBgNVHSMEGDAWgBT9f1xhI06cdr9f\nhizBFmmQlQQN0DAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQA+\nZ6i9ZdOMBUA0yYvE7xVVNiMwV+7N0cnL05JR6KuvDRkVL4j1l2D4tGSmP1AsD3I8\n4ozLPtQ0Dk407KSN/X3Tg+vd4yHefEq4yOZlHbEV+YcDgix8svvPLA8B9lSuhnne\nZ4byRYaSAhhMiiEBQTMBjaAIjcFiif3g079NbeGopQjrvUTR67Hb8JfmIhZ6icbh\n5VoihAPsRno3Tjg+5z7E1nEY3zW+VK9rp1HqFIDazwAQSUFpM/GVn+HemMS7optR\njEMhkqAS4E/+GUGAxWhfeQC4kZKJL2JgiNaCghEugGY7UWemXGPRcgN87UtTLqg6\n6FW15bRjDzogPB6Kell5\n-----END CERTIFICATE-----\n"
    }
   ],
   "trustAnchors": [
    {
     "pemCertificate": "-----BEGIN CERTIFICATE-----\nMIIDazCCAlOgAwIBAgIUfCJocHZRYbgzWUuaP6j9HM4Fp9YwDQYJKoZIhvcNAQEL\nBQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM\nGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMzAzMjExNjM2MDFaFw0yNTEy\nMTYxNjM2MDFaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw\nHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggEiMA0GCSqGSIb3DQEB\nAQUAA4IBDwAwggEKAoIBAQC/XVbAB3l6cnizpofzSJ5o4Iko3WgNV1PNeTe/JvZP\nzSoJC22TY3foqrtM0YBdNWfBEKyT5bTGYLJ67P9v2CX4xfUtQHiuWkF1OaA8/umb\nvzaQt2V1XsPC4nlnByeL8nT9wtLEwZ/8qeYJ0c0PS+1zJ5kEVRCvTg3ay6o7VZv5\nt3TvaQBjNwAhaUlFw9R6tqCG7Ztt+4d8dxCV4LjC6ygHtzjfmJQJBgMJcCB3F0ch\nFMGFz9I4OGXrqukzuSJLMg/Nok3EkOV7b4vptdICfzJJTDfvyQ9radzC+O7mtnRp\nuYGdTFKbDOglwxliyJJuQ+GAWjaW2sM51EM6AIk88bylAgMBAAGjUzBRMB0GA1Ud\nDgQWBBT9f1xhI06cdr9fhizBFmmQlQQN0DAfBgNVHSMEGDAWgBT9f1xhI06cdr9f\nhizBFmmQlQQN0DAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQA+\nZ6i9ZdOMBUA0yYvE7xVVNiMwV+7N0cnL05JR6KuvDRkVL4j1l2D4tGSmP1AsD3I8\n4ozLPtQ0Dk407KSN/X3Tg+vd4yHefEq4yOZlHbEV+YcDgix8svvPLA8B9lSuhnne\nZ4byRYaSAhhMiiEBQTMBjaAIjcFiif3g079NbeGopQjrvUTR67Hb8JfmIhZ6icbh\n5VoihAPsRno3Tjg+5z7E1nEY3zW+VK9rp1HqFIDazwAQSUFpM/GVn+HemMS7optR\njEMhkqAS4E/+GUGAxWhfeQC4kZKJL2JgiNaCghEugGY7UWemXGPRcgN87UtTLqg6\n6FW15bRjDzogPB6Kell5\n-----END CERTIFICATE-----\n"
    }
   ]
  }
 ]
}

-----------------------------------------------------
2023/08/16 00:36:15 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 400 Bad Request
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Tue, 15 Aug 2023 22:36:15 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "error": {
    "code": 400,
    "message": "update_mask should be set to \"*\", other masks are not supported yet",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.BadRequest"
      }
    ]
  }
}

```

Reply:
And I cannot seems to update the labels, the response of the patch call is of type `operation` and has the changed labels in it but the actual resource hasn't been updated. I marked them for now as `immutable` but is that something we can ask with the API team?

**1. How the resource starts**

```
---[ REQUEST ]---------------------------------------
GET /v1/projects/project-x/locations/us-central1/trustConfigs/tf-test-trust-config8qh2jw0aww?alt=json HTTP/1.1
Host: certificatemanager.googleapis.com
User-Agent: Terraform/1.5.0 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google/dev
Content-Type: application/json
Accept-Encoding: gzip


-----------------------------------------------------
2023/08/16 01:27:42 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Tue, 15 Aug 2023 23:27:42 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "name": "projects/project-x/locations/us-central1/trustConfigs/tf-test-trust-config8qh2jw0aww",
  "createTime": "2023-08-15T23:27:25.531813072Z",
  "updateTime": "2023-08-15T23:27:26.750787205Z",
  "labels": {
    "foo": "bar"
  },
  "description": "sample description for the trust config",
  "etag": "kTdB9oRbqTEa3f93Wzb9xdeMZmWmYtLpVWzCrpjfc6g",
  "trustStores": [
    {
      "trustAnchors": [
        {
          "pemCertificate": "-----BEGIN CERTIFICATE-----\nMIIDDzCCAfegAwIBAgIUDOiCLH9QNMMYnjPZVf4VwO9blsEwDQYJKoZIhvcNAQEL\nBQAwFjEUMBIGA1UEAwwLZXhhbXBsZS5jb20wIBcNMjIwODI0MDg0MDUxWhgPMzAy\nMTEyMjUwODQwNTFaMBYxFDASBgNVBAMMC2V4YW1wbGUuY29tMIIBIjANBgkqhkiG\n9w0BAQEFAAOCAQ8AMIIBCgKCAQEAvOT925GG4lKV9HvAHsbecMhGPAqjhVRC26iZ\nUJC8oSWOu95lWJSX5ZhbiF6Nz192wDGV/VAh3Lxj8RYtcn75eDxQKTcKouDld+To\nCGIStPFWbR6rbysLuZqFVEXVOTvp2QIegInfrvnGC4j7Qpic7zrFB9HzJx+0HpeE\nyO4gkdzJfEK/gMmolUgJrKX59o+0+Rj+Jq3EtcQxL1fVBVJSx0NvpoR1eYpnHMr/\nrJKZkUUZ2xE86hrtpiP6OEYQTi00rmf4GnZF5QfGGD0xuoQXtR7Tu+XhKibXIhxc\nD4RzPLX1QS040PXvmMPLDb4YlUQ6V3Rs42JDvkkDwIMXZvn8awIDAQABo1MwUTAd\nBgNVHQ4EFgQURuo1CCZZAUv7xi02f2nC5tRbf18wHwYDVR0jBBgwFoAURuo1CCZZ\nAUv7xi02f2nC5tRbf18wDwYDVR0TAQH/BAUwAwEB/zANBgkqhkiG9w0BAQsFAAOC\nAQEAqx3tDxurnYr9EUPhF5/LlDPYM+VI7EgrKdRnuIqUlZI0tm3vOGME0te6dBTC\nYLNaHLW3m/4Tm4M2eg0Kpz6CxJfn3109G31dCi0xwzSDHf5TPUWvqIVhq5WRgMIf\nn8KYBlQSmqdJBRztUIQH/UPFnSbxymlS4s5qwDgTH5ag9EEBcnWsQ2LZjKi0eqve\nMaqAvvB+j8RGZzYY4re94bSJI42zIZ6nMWPtXwRuDc30xl/u+E0jWIgWbPwSd6Km\n3wnJnGiU2ezPGq3zEU+Rc39VVIFKQpciNeYuF3neHPJvYOf58qW2Z8s0VH0MR1x3\n3DoO/e30FIr9j+PRD+s5BPKF2A==\n-----END CERTIFICATE-----\n"
        }
      ],
      "intermediateCas": [
        {
          "pemCertificate": "-----BEGIN CERTIFICATE-----\nMIIDDzCCAfegAwIBAgIUDOiCLH9QNMMYnjPZVf4VwO9blsEwDQYJKoZIhvcNAQEL\nBQAwFjEUMBIGA1UEAwwLZXhhbXBsZS5jb20wIBcNMjIwODI0MDg0MDUxWhgPMzAy\nMTEyMjUwODQwNTFaMBYxFDASBgNVBAMMC2V4YW1wbGUuY29tMIIBIjANBgkqhkiG\n9w0BAQEFAAOCAQ8AMIIBCgKCAQEAvOT925GG4lKV9HvAHsbecMhGPAqjhVRC26iZ\nUJC8oSWOu95lWJSX5ZhbiF6Nz192wDGV/VAh3Lxj8RYtcn75eDxQKTcKouDld+To\nCGIStPFWbR6rbysLuZqFVEXVOTvp2QIegInfrvnGC4j7Qpic7zrFB9HzJx+0HpeE\nyO4gkdzJfEK/gMmolUgJrKX59o+0+Rj+Jq3EtcQxL1fVBVJSx0NvpoR1eYpnHMr/\nrJKZkUUZ2xE86hrtpiP6OEYQTi00rmf4GnZF5QfGGD0xuoQXtR7Tu+XhKibXIhxc\nD4RzPLX1QS040PXvmMPLDb4YlUQ6V3Rs42JDvkkDwIMXZvn8awIDAQABo1MwUTAd\nBgNVHQ4EFgQURuo1CCZZAUv7xi02f2nC5tRbf18wHwYDVR0jBBgwFoAURuo1CCZZ\nAUv7xi02f2nC5tRbf18wDwYDVR0TAQH/BAUwAwEB/zANBgkqhkiG9w0BAQsFAAOC\nAQEAqx3tDxurnYr9EUPhF5/LlDPYM+VI7EgrKdRnuIqUlZI0tm3vOGME0te6dBTC\nYLNaHLW3m/4Tm4M2eg0Kpz6CxJfn3109G31dCi0xwzSDHf5TPUWvqIVhq5WRgMIf\nn8KYBlQSmqdJBRztUIQH/UPFnSbxymlS4s5qwDgTH5ag9EEBcnWsQ2LZjKi0eqve\nMaqAvvB+j8RGZzYY4re94bSJI42zIZ6nMWPtXwRuDc30xl/u+E0jWIgWbPwSd6Km\n3wnJnGiU2ezPGq3zEU+Rc39VVIFKQpciNeYuF3neHPJvYOf58qW2Z8s0VH0MR1x3\n3DoO/e30FIr9j+PRD+s5BPKF2A==\n-----END CERTIFICATE-----\n"
        }
      ]
    }
  ]
}
```

**2. The patch call to change the labels**

```
---[ REQUEST ]---------------------------------------
PATCH /v1/projects/project-x/locations/us-central1/trustConfigs/tf-test-trust-config8qh2jw0aww?alt=json&updateMask=%2A HTTP/1.1
Host: certificatemanager.googleapis.com
User-Agent: Terraform/1.5.0 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google/dev
Content-Length: 2714
Content-Type: application/json
Accept-Encoding: gzip

{
 "description": "sample description for the trust config 2",
 "labels": {
  "bar": "foo"
 },
 "trustStores": [
  {
   "intermediateCas": [
    {
     "pemCertificate": "-----BEGIN CERTIFICATE-----\nMIIDazCCAlOgAwIBAgIUfCJocHZRYbgzWUuaP6j9HM4Fp9YwDQYJKoZIhvcNAQEL\nBQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM\nGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMzAzMjExNjM2MDFaFw0yNTEy\nMTYxNjM2MDFaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw\nHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggEiMA0GCSqGSIb3DQEB\nAQUAA4IBDwAwggEKAoIBAQC/XVbAB3l6cnizpofzSJ5o4Iko3WgNV1PNeTe/JvZP\nzSoJC22TY3foqrtM0YBdNWfBEKyT5bTGYLJ67P9v2CX4xfUtQHiuWkF1OaA8/umb\nvzaQt2V1XsPC4nlnByeL8nT9wtLEwZ/8qeYJ0c0PS+1zJ5kEVRCvTg3ay6o7VZv5\nt3TvaQBjNwAhaUlFw9R6tqCG7Ztt+4d8dxCV4LjC6ygHtzjfmJQJBgMJcCB3F0ch\nFMGFz9I4OGXrqukzuSJLMg/Nok3EkOV7b4vptdICfzJJTDfvyQ9radzC+O7mtnRp\nuYGdTFKbDOglwxliyJJuQ+GAWjaW2sM51EM6AIk88bylAgMBAAGjUzBRMB0GA1Ud\nDgQWBBT9f1xhI06cdr9fhizBFmmQlQQN0DAfBgNVHSMEGDAWgBT9f1xhI06cdr9f\nhizBFmmQlQQN0DAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQA+\nZ6i9ZdOMBUA0yYvE7xVVNiMwV+7N0cnL05JR6KuvDRkVL4j1l2D4tGSmP1AsD3I8\n4ozLPtQ0Dk407KSN/X3Tg+vd4yHefEq4yOZlHbEV+YcDgix8svvPLA8B9lSuhnne\nZ4byRYaSAhhMiiEBQTMBjaAIjcFiif3g079NbeGopQjrvUTR67Hb8JfmIhZ6icbh\n5VoihAPsRno3Tjg+5z7E1nEY3zW+VK9rp1HqFIDazwAQSUFpM/GVn+HemMS7optR\njEMhkqAS4E/+GUGAxWhfeQC4kZKJL2JgiNaCghEugGY7UWemXGPRcgN87UtTLqg6\n6FW15bRjDzogPB6Kell5\n-----END CERTIFICATE-----\n"
    }
   ],
   "trustAnchors": [
    {
     "pemCertificate": "-----BEGIN CERTIFICATE-----\nMIIDazCCAlOgAwIBAgIUfCJocHZRYbgzWUuaP6j9HM4Fp9YwDQYJKoZIhvcNAQEL\nBQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM\nGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMzAzMjExNjM2MDFaFw0yNTEy\nMTYxNjM2MDFaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw\nHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggEiMA0GCSqGSIb3DQEB\nAQUAA4IBDwAwggEKAoIBAQC/XVbAB3l6cnizpofzSJ5o4Iko3WgNV1PNeTe/JvZP\nzSoJC22TY3foqrtM0YBdNWfBEKyT5bTGYLJ67P9v2CX4xfUtQHiuWkF1OaA8/umb\nvzaQt2V1XsPC4nlnByeL8nT9wtLEwZ/8qeYJ0c0PS+1zJ5kEVRCvTg3ay6o7VZv5\nt3TvaQBjNwAhaUlFw9R6tqCG7Ztt+4d8dxCV4LjC6ygHtzjfmJQJBgMJcCB3F0ch\nFMGFz9I4OGXrqukzuSJLMg/Nok3EkOV7b4vptdICfzJJTDfvyQ9radzC+O7mtnRp\nuYGdTFKbDOglwxliyJJuQ+GAWjaW2sM51EM6AIk88bylAgMBAAGjUzBRMB0GA1Ud\nDgQWBBT9f1xhI06cdr9fhizBFmmQlQQN0DAfBgNVHSMEGDAWgBT9f1xhI06cdr9f\nhizBFmmQlQQN0DAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQA+\nZ6i9ZdOMBUA0yYvE7xVVNiMwV+7N0cnL05JR6KuvDRkVL4j1l2D4tGSmP1AsD3I8\n4ozLPtQ0Dk407KSN/X3Tg+vd4yHefEq4yOZlHbEV+YcDgix8svvPLA8B9lSuhnne\nZ4byRYaSAhhMiiEBQTMBjaAIjcFiif3g079NbeGopQjrvUTR67Hb8JfmIhZ6icbh\n5VoihAPsRno3Tjg+5z7E1nEY3zW+VK9rp1HqFIDazwAQSUFpM/GVn+HemMS7optR\njEMhkqAS4E/+GUGAxWhfeQC4kZKJL2JgiNaCghEugGY7UWemXGPRcgN87UtTLqg6\n6FW15bRjDzogPB6Kell5\n-----END CERTIFICATE-----\n"
    }
   ]
  }
 ]
}
```

**3. Response from `operation` when `done` is `true`, (shows the changed labels)**

```
---[ REQUEST ]---------------------------------------
GET /v1/projects/project-x/locations/us-central1/operations/operation-1692142063986-602fe86a22559-ef00a4ad-d97b7dc2?alt=json HTTP/1.1
Host: certificatemanager.googleapis.com
User-Agent: Terraform/1.5.0 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google/dev
Content-Type: application/json
Accept-Encoding: gzip


-----------------------------------------------------
2023/08/16 01:27:55 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Tue, 15 Aug 2023 23:27:55 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "name": "projects/project-x/locations/us-central1/operations/operation-1692142063986-602fe86a22559-ef00a4ad-d97b7dc2",
  "metadata": {
    "@type": "type.googleapis.com/google.cloud.certificatemanager.v1.OperationMetadata",
    "createTime": "2023-08-15T23:27:43.996869339Z",
    "endTime": "2023-08-15T23:27:45.376116909Z",
    "target": "projects/project-x/locations/us-central1/trustConfigs/tf-test-trust-config8qh2jw0aww",
    "verb": "update",
    "requestedCancellation": false,
    "apiVersion": "v1"
  },
  "done": true,
  "response": {
    "@type": "type.googleapis.com/google.cloud.certificatemanager.v1.TrustConfig",
    "name": "projects/pproject-x/locations/us-central1/trustConfigs/tf-test-trust-config8qh2jw0aww",
    "createTime": "2023-08-15T23:27:25.531813072Z",
    "updateTime": "2023-08-15T23:27:43.999649039Z",
    "labels": {
      "bar": "foo"
    },
    "description": "sample description for the trust config 2",
    "etag": "Km2AxX3pQyeOWqf9G-4nFCvXjTFKEBvTpRfiPdf_Gr8",
    "trustStores": [
      {
        "trustAnchors": [
          {
            "pemCertificate": "-----BEGIN CERTIFICATE-----\nMIIDazCCAlOgAwIBAgIUfCJocHZRYbgzWUuaP6j9HM4Fp9YwDQYJKoZIhvcNAQEL\nBQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM\nGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMzAzMjExNjM2MDFaFw0yNTEy\nMTYxNjM2MDFaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw\nHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggEiMA0GCSqGSIb3DQEB\nAQUAA4IBDwAwggEKAoIBAQC/XVbAB3l6cnizpofzSJ5o4Iko3WgNV1PNeTe/JvZP\nzSoJC22TY3foqrtM0YBdNWfBEKyT5bTGYLJ67P9v2CX4xfUtQHiuWkF1OaA8/umb\nvzaQt2V1XsPC4nlnByeL8nT9wtLEwZ/8qeYJ0c0PS+1zJ5kEVRCvTg3ay6o7VZv5\nt3TvaQBjNwAhaUlFw9R6tqCG7Ztt+4d8dxCV4LjC6ygHtzjfmJQJBgMJcCB3F0ch\nFMGFz9I4OGXrqukzuSJLMg/Nok3EkOV7b4vptdICfzJJTDfvyQ9radzC+O7mtnRp\nuYGdTFKbDOglwxliyJJuQ+GAWjaW2sM51EM6AIk88bylAgMBAAGjUzBRMB0GA1Ud\nDgQWBBT9f1xhI06cdr9fhizBFmmQlQQN0DAfBgNVHSMEGDAWgBT9f1xhI06cdr9f\nhizBFmmQlQQN0DAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQA+\nZ6i9ZdOMBUA0yYvE7xVVNiMwV+7N0cnL05JR6KuvDRkVL4j1l2D4tGSmP1AsD3I8\n4ozLPtQ0Dk407KSN/X3Tg+vd4yHefEq4yOZlHbEV+YcDgix8svvPLA8B9lSuhnne\nZ4byRYaSAhhMiiEBQTMBjaAIjcFiif3g079NbeGopQjrvUTR67Hb8JfmIhZ6icbh\n5VoihAPsRno3Tjg+5z7E1nEY3zW+VK9rp1HqFIDazwAQSUFpM/GVn+HemMS7optR\njEMhkqAS4E/+GUGAxWhfeQC4kZKJL2JgiNaCghEugGY7UWemXGPRcgN87UtTLqg6\n6FW15bRjDzogPB6Kell5\n-----END CERTIFICATE-----\n"
          }
        ],
        "intermediateCas": [
          {
            "pemCertificate": "-----BEGIN CERTIFICATE-----\nMIIDazCCAlOgAwIBAgIUfCJocHZRYbgzWUuaP6j9HM4Fp9YwDQYJKoZIhvcNAQEL\nBQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM\nGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMzAzMjExNjM2MDFaFw0yNTEy\nMTYxNjM2MDFaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw\nHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggEiMA0GCSqGSIb3DQEB\nAQUAA4IBDwAwggEKAoIBAQC/XVbAB3l6cnizpofzSJ5o4Iko3WgNV1PNeTe/JvZP\nzSoJC22TY3foqrtM0YBdNWfBEKyT5bTGYLJ67P9v2CX4xfUtQHiuWkF1OaA8/umb\nvzaQt2V1XsPC4nlnByeL8nT9wtLEwZ/8qeYJ0c0PS+1zJ5kEVRCvTg3ay6o7VZv5\nt3TvaQBjNwAhaUlFw9R6tqCG7Ztt+4d8dxCV4LjC6ygHtzjfmJQJBgMJcCB3F0ch\nFMGFz9I4OGXrqukzuSJLMg/Nok3EkOV7b4vptdICfzJJTDfvyQ9radzC+O7mtnRp\nuYGdTFKbDOglwxliyJJuQ+GAWjaW2sM51EM6AIk88bylAgMBAAGjUzBRMB0GA1Ud\nDgQWBBT9f1xhI06cdr9fhizBFmmQlQQN0DAfBgNVHSMEGDAWgBT9f1xhI06cdr9f\nhizBFmmQlQQN0DAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQA+\nZ6i9ZdOMBUA0yYvE7xVVNiMwV+7N0cnL05JR6KuvDRkVL4j1l2D4tGSmP1AsD3I8\n4ozLPtQ0Dk407KSN/X3Tg+vd4yHefEq4yOZlHbEV+YcDgix8svvPLA8B9lSuhnne\nZ4byRYaSAhhMiiEBQTMBjaAIjcFiif3g079NbeGopQjrvUTR67Hb8JfmIhZ6icbh\n5VoihAPsRno3Tjg+5z7E1nEY3zW+VK9rp1HqFIDazwAQSUFpM/GVn+HemMS7optR\njEMhkqAS4E/+GUGAxWhfeQC4kZKJL2JgiNaCghEugGY7UWemXGPRcgN87UtTLqg6\n6FW15bRjDzogPB6Kell5\n-----END CERTIFICATE-----\n"
          }
        ]
      }
    ]
  }
}
```

**4. Another GET request and this shows the old labels**

```
---[ REQUEST ]---------------------------------------
GET /v1/projects/project-x/locations/us-central1/trustConfigs/tf-test-trust-config8qh2jw0aww?alt=json HTTP/1.1
Host: certificatemanager.googleapis.com
User-Agent: Terraform/1.5.0 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google/dev
Content-Type: application/json
Accept-Encoding: gzip


-----------------------------------------------------
2023/08/16 01:27:56 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Tue, 15 Aug 2023 23:27:56 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "name": "projects/project-x/locations/us-central1/trustConfigs/tf-test-trust-config8qh2jw0aww",
  "createTime": "2023-08-15T23:27:25.531813072Z",
  "updateTime": "2023-08-15T23:27:45.398536388Z",
  "labels": {
    "foo": "bar"
  },
  "description": "sample description for the trust config 2",
  "etag": "3AovWQ2feX9_dH-cUwW6qnd-h4d_8xASlBk16cZ6s9I",
  "trustStores": [
    {
      "trustAnchors": [
        {
          "pemCertificate": "-----BEGIN CERTIFICATE-----\nMIIDazCCAlOgAwIBAgIUfCJocHZRYbgzWUuaP6j9HM4Fp9YwDQYJKoZIhvcNAQEL\nBQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM\nGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMzAzMjExNjM2MDFaFw0yNTEy\nMTYxNjM2MDFaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw\nHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggEiMA0GCSqGSIb3DQEB\nAQUAA4IBDwAwggEKAoIBAQC/XVbAB3l6cnizpofzSJ5o4Iko3WgNV1PNeTe/JvZP\nzSoJC22TY3foqrtM0YBdNWfBEKyT5bTGYLJ67P9v2CX4xfUtQHiuWkF1OaA8/umb\nvzaQt2V1XsPC4nlnByeL8nT9wtLEwZ/8qeYJ0c0PS+1zJ5kEVRCvTg3ay6o7VZv5\nt3TvaQBjNwAhaUlFw9R6tqCG7Ztt+4d8dxCV4LjC6ygHtzjfmJQJBgMJcCB3F0ch\nFMGFz9I4OGXrqukzuSJLMg/Nok3EkOV7b4vptdICfzJJTDfvyQ9radzC+O7mtnRp\nuYGdTFKbDOglwxliyJJuQ+GAWjaW2sM51EM6AIk88bylAgMBAAGjUzBRMB0GA1Ud\nDgQWBBT9f1xhI06cdr9fhizBFmmQlQQN0DAfBgNVHSMEGDAWgBT9f1xhI06cdr9f\nhizBFmmQlQQN0DAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQA+\nZ6i9ZdOMBUA0yYvE7xVVNiMwV+7N0cnL05JR6KuvDRkVL4j1l2D4tGSmP1AsD3I8\n4ozLPtQ0Dk407KSN/X3Tg+vd4yHefEq4yOZlHbEV+YcDgix8svvPLA8B9lSuhnne\nZ4byRYaSAhhMiiEBQTMBjaAIjcFiif3g079NbeGopQjrvUTR67Hb8JfmIhZ6icbh\n5VoihAPsRno3Tjg+5z7E1nEY3zW+VK9rp1HqFIDazwAQSUFpM/GVn+HemMS7optR\njEMhkqAS4E/+GUGAxWhfeQC4kZKJL2JgiNaCghEugGY7UWemXGPRcgN87UtTLqg6\n6FW15bRjDzogPB6Kell5\n-----END CERTIFICATE-----\n"
        }
      ],
      "intermediateCas": [
        {
          "pemCertificate": "-----BEGIN CERTIFICATE-----\nMIIDazCCAlOgAwIBAgIUfCJocHZRYbgzWUuaP6j9HM4Fp9YwDQYJKoZIhvcNAQEL\nBQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM\nGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMzAzMjExNjM2MDFaFw0yNTEy\nMTYxNjM2MDFaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw\nHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggEiMA0GCSqGSIb3DQEB\nAQUAA4IBDwAwggEKAoIBAQC/XVbAB3l6cnizpofzSJ5o4Iko3WgNV1PNeTe/JvZP\nzSoJC22TY3foqrtM0YBdNWfBEKyT5bTGYLJ67P9v2CX4xfUtQHiuWkF1OaA8/umb\nvzaQt2V1XsPC4nlnByeL8nT9wtLEwZ/8qeYJ0c0PS+1zJ5kEVRCvTg3ay6o7VZv5\nt3TvaQBjNwAhaUlFw9R6tqCG7Ztt+4d8dxCV4LjC6ygHtzjfmJQJBgMJcCB3F0ch\nFMGFz9I4OGXrqukzuSJLMg/Nok3EkOV7b4vptdICfzJJTDfvyQ9radzC+O7mtnRp\nuYGdTFKbDOglwxliyJJuQ+GAWjaW2sM51EM6AIk88bylAgMBAAGjUzBRMB0GA1Ud\nDgQWBBT9f1xhI06cdr9fhizBFmmQlQQN0DAfBgNVHSMEGDAWgBT9f1xhI06cdr9f\nhizBFmmQlQQN0DAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQA+\nZ6i9ZdOMBUA0yYvE7xVVNiMwV+7N0cnL05JR6KuvDRkVL4j1l2D4tGSmP1AsD3I8\n4ozLPtQ0Dk407KSN/X3Tg+vd4yHefEq4yOZlHbEV+YcDgix8svvPLA8B9lSuhnne\nZ4byRYaSAhhMiiEBQTMBjaAIjcFiif3g079NbeGopQjrvUTR67Hb8JfmIhZ6icbh\n5VoihAPsRno3Tjg+5z7E1nEY3zW+VK9rp1HqFIDazwAQSUFpM/GVn+HemMS7optR\njEMhkqAS4E/+GUGAxWhfeQC4kZKJL2JgiNaCghEugGY7UWemXGPRcgN87UtTLqg6\n6FW15bRjDzogPB6Kell5\n-----END CERTIFICATE-----\n"
        }
      ]
    }
  ]
}
```

Reply:
Weird ! I wonder if this is a eventual consistency issue ( ie change is in the backend but takes time to reflect)

Bizarre update behavior all around with the api not accepting other `updateMask`(s)

Reply:
I had to test it and even after a few minutes the changes weren't reflected, either it takes a long time for the changes to reflect or there is something else going on.